src/main/java/org/wildfly/security/auth/realm/ldap/LdapSecurityRealm.java

```diff
@@ -637,124 +630,91 @@ private LdapIdentity getIdentity() throws RealmUnavailableException {
             }
         }
 
-        private Map<String, Collection<String>> extractFilteredAttributes(SearchResult result, DirContext context) {
-            String principalDn = result.getNameInNamespace();
-
-            return extractAttributes(attribute -> attribute.getFilter() != null, attribute -> {
-                Collection<String> values = new ArrayList<>();
-
-                String searchDn = attribute.getSearchDn();
-
-                if (searchDn == null) {
-                    searchDn = identityMapping.searchDn;
+        private String valueFromDn(AttributeMapping mapping, final String dn) {
+            String valueRdn = mapping.getRdn();
+            try {
+                for (Rdn rdn : new LdapName(dn).getRdns()) {
+                    if (rdn.getType().equalsIgnoreCase(valueRdn)) {
+                        return rdn.getValue().toString();
+                    }
                 }
+            } catch (Exception cause) {
+                throw log.ldapRealmInvalidRdnForAttribute(mapping.getName(), dn, valueRdn);
```

Comment:
If so, then it's a good idea to rename it to "ignored", which has the added benefit that IDEA will not report it as unused in this case.